contracts/strategies/modules/recipient/IRecipientEligibilityModule.sol

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.19;
+
+import {IStrategy} from "../../IStrategy.sol";
+
+interface IRecipientEligibilityModule {
```

Comment:
We will still have situations where certain strategies will want to use different statuses. What's the best way to handle it in this architecture?

Reply:
@nfrgosselin I think that's fine. They can add any logic they want to this. They just need to expose the `getRecipientStatus()` function that returns one of the status from our enum. 

They can track internally however they want, and expose additional functions with more options, etc.

The point is that there needs to be one function that simplifies into these four states, as that's what our frontend will be able to read and know will always be consistent.